pkg/spec/spec.go

```diff
@@ -31,14 +31,17 @@ type Function struct {
 
 // FunctionSpec contains func specification
 type FunctionSpec struct {
-	Handler  string             `json:"handler"`
-	Function string             `json:"function"`
-	Runtime  string             `json:"runtime"`
-	Type     string             `json:"type"`
-	Topic    string             `json:"topic"`
-	Schedule string             `json:"schedule"`
-	Deps     string             `json:"deps"`
-	Template v1.PodTemplateSpec `json:"template" protobuf:"bytes,3,opt,name=template"`
+	Handler     string             `json:"handler"`      // Function handler: "file.function"
+	Function    string             `json:"function"`     // Function file content or URL of the function
+	ContentType string             `json:"content-type"` // Function file content type
+	File        string             `json:"file"`         // File name that contains the function
```

Comment:
file name relative to what?

Reply:
It is just the filename of the file when specified with --from-file. It is used to retrieve the correct file from the mounted volume (since the name in the `handler` may not be the same).